### PR TITLE
Set torch device from commandline

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
The torch device string can now be set from the commandline. The default behavior is the same as before: it will use cuda if it's detected, or cpu otherwise. This allows overriding the default logic (e.g. force cpu even if you support cuda), or specifying a device index if e.g. you have two different GPU types.

Also logs the device type so that it's obvious which is being used.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://forum.unity.com/threads/ml-agent-server-build-headless-with-visual-observations-and-multiple-gpus.1042456/
https://github.com/Unity-Technologies/ml-agents/issues/4876

### Types of change(s)
- [x] New feature

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/cli_utils.py [hunk 1]
```diff
@@ -252,6 +252,15 @@ def _create_parser() -> argparse.ArgumentParser:
         help="Whether to run the Unity executable in no-graphics mode (i.e. without initializing "
         "the graphics driver. Use this only if your agents don't use visual observations.",
     )
+
+    torch_conf = argparser.add_argument_group(title="Torch Configuration")
```
Comment (dongruoping) [inline ml-agents/mlagents/trainers/cli_utils.py:256]: `--cpu` option should be moved into this group or just integrate with the device argument

**Review by dongruoping — COMMENTED** [review]
It would be good to also mention this option somewhere in to doc

File: com.unity.ml-agents/CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,9 @@ removed when training with a player. The Editor still requires it to be clamped
   Changed the namespace and file names of classes in com.unity.ml-agents.extensions. (#4849)
 
 #### ml-agents / ml-agents-envs / gym-unity (Python)
+- Added a `--torch-device` commandline option to `mlagent-learn`, which sets the default
```
Comment (ervteng) [inline com.unity.ml-agents/CHANGELOG.md:28]:
```suggestion
- Added a `--torch-device` commandline option to `mlagents-learn`, which sets the default
```

Comment (chriselion) [reply com.unity.ml-agents/CHANGELOG.md]: Fixed

File: ml-agents/mlagents/trainers/cli_utils.py (hunk 1 above)
Comment (chriselion) [reply ml-agents/mlagents/trainers/cli_utils.py:250]: Removed `--cpu` since it's obsolete (and added a note in the changelog)

**Review by ervteng — APPROVED** [review]
:shipit:

**Pull request merged**